src/libraries/System.Private.CoreLib/src/System/IO/Directory.cs

```diff
@@ -26,7 +26,7 @@ public static partial class Directory
             if (path.Length == 0)
                 throw new ArgumentException(SR.Argument_PathEmpty, nameof(path));
 
-            string fullPath = Path.GetFullPath(path);
+            string fullPath = Path.GetFullyQualifiedPath(path);
```

Comment:
This GetFullyQualifiedPath helper doesn't do all the checks GetFullPath does: GetFullPath checks not only for a null or empty string, but also for a string containing \0, and on Windows, GetFullPath doesn't just validate Length = 0, it also validates that Length != 0 then isn't just composed of all spaces.

As such, most if not all of the substitutions in this PR aren't valid.

Reply:
I'll add that check to the `GetFullyQualifiedPath` method. I'll then remove that check in the methods that call it.